test-util/src/main/java/io/camunda/zeebe/test/util/stream/StreamWrapper.java

```diff
@@ -52,12 +53,25 @@ public S skipUntil(final Predicate<T> predicate) {
    * short-circuiting operation; limits the stream to the first element that fulfills the predicate
    */
   public S limit(final Predicate<T> predicate) {
+    return limitByCount(predicate, 1);
+  }
+
+  /**
+   * short-circuiting operation; limits the stream until a given count of elements has matched the
+   * predicate.
+   */
+  public S limitByCount(final Predicate<T> predicate, final int count) {
     // looks a bit hacky but we can't use takeWhile(predicate.negate()) here
     // because this doesn't include the element that fulfills the predicate
     // and the underlying record iterator blocks on hasNext()
 
     final var records = new ArrayList<T>();
-    wrappedStream.peek(records::add).filter(predicate).findFirst();
+    final var index = new AtomicInteger();
+    wrappedStream
+        .peek(records::add)
+        .filter(record -> predicate.test(record) && index.getAndIncrement() < count)
+        .limit(count)
+        .count();
```

Comment:
I was thinking a bit about this, because `count`'s result is unused, and `peek` is used outside of [its intended scope](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#peek-java.util.function.Consumer-).

>This method exists mainly to support debugging, where you want to see the elements as they flow past a certain point in a pipeline

❓ What do you think about this alternative to inclusive take while?
```java
final var index = new AtomicInteger();
final var hit = new AtomicBoolean();
return supply(
    wrappedStream.filter(
        e -> {
          if (hit.get()) {
            return false;
          }

          hit.set(predicate.test(e) && index.getAndIncrement() < count);
          // even when the predicate matches we still need to include the current element
          return true;
        }));
```

Reply:
I understand why you would say this. I also struggled a bit with different solutions.

Problem is, our wrapped stream wait by default for 5 seconds, so just using this filter isn't sufficient. We need to have some limiting factor to short-circuit the stream, like we do in all our tests. 

The problem here is what do we limit it on. We don't know how many records are expected to be returned. This is why the "dirty" peek workaround is used. We can limit the stream once we've filtered `count` amount of elements. The peek makes sure we get all the records before this as well.
`limit` on it's own doesn't work as it's an intermediate operation of the stream. We need to end on a terminal operation, which is why I use `count` here.

Without all this all tests using the `limit` will run >5 seconds.